torch/functional.py

```diff
@@ -361,7 +371,17 @@ def parse_subscript(n: int) -> str:
         # in the original implementation this line is omitted
         return einsum(equation, *_operands)
 
-    return _VF.einsum(equation, operands)  # type: ignore[attr-defined]
+    if len(operands) <= 2:
+        # the optimal path for contracting 0 or 1 time(s) is already optimized
+        return _VF.einsum(equation, operands)  # type: ignore[attr-defined]
+
+    flattened_path = None
+    if HAS_OPT_EINSUM:
+        path = contract_path(equation, *operands)[0]
```

Comment:
nit: just for it to be more clear that contract_path is from opt_einsum, we could just `import opt_einsum` and then do `opt_einsum.contract_path`.